src/client/Microsoft.Identity.Client/ManagedIdentity/AzureArcManagedIdentitySource.cs

```diff
@@ -69,6 +71,45 @@ protected override ManagedIdentityRequest CreateRequest(string resource)
             return request;
         }
 
+        protected override async Task<ManagedIdentityResponse> HandleResponseAsync(
```

Comment:
Seems like there is a bit of duplicate code between the two overloads of `HandleResponseAsync`. Can you consilidate the logic?